main.py

```
@@ -0,0 +1,162 @@
+import sys
+import queue
+import re
+import urllib.parse
+import requests # this package needs to be installed
+
+# NOTES:
+#       uses python 3
+#
+#       because this is a command line utility, the code looks like a script
+#       the only 2 useful improvements are UlrQueue class in case the number of links is large
+#       and a comment about parallel processing
+#
+#       main method is in the bottom
+
+# class that hides the implementation of the url queue in case the queue should be large
+class UrlQueue:
+
+    def __init__(self):
+        self._q = queue.Queue(maxsize=max_queue_size)
+        self._cache = set()
+
+    def size(self):
+        return self._q.qsize()
+
+    # to skip processing a url
+    def skip(self, url):
+        self._cache.add(url)
+
+    def put(self, url):
+        try:
+            if not url in self._cache:
+                self._cache.add(url)
+                self._q.put(url, block=False)
+        except:
+            # in this implementation is memory is filled, there is nothing to do
+            # no need to print errors because this is by design
+            pass
```

I'm not quite clear on the intent of the `except: pass` here. What's the exception that you're trying to catch? I'd prefer to be explicit about it, instead of catching everything.